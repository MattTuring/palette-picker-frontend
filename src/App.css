.App {
  display: grid;
  height: 100vh;
  grid-template-rows: 1fr 11fr;
  grid-template-columns: 3fr 1fr;
  grid-template-areas: 'nav nav'
                       'color-picker projects'
                       'palettes palettes'
}

nav {
  grid-area: nav;
  background: -webkit-linear-gradient(-180deg, rgb(30, 251, 129), rgb(229, 2, 212), rgb(95, 6, 192));
  text-align: center;
  font-size: 70px;
  box-shadow: 2px 1px 8px #000000c9;
}

.color-picker {
  display: flex;
  flex-direction: row;
  justify-content: center;
  grid-area: color-picker;
  padding: 40px;
}

.projects {
  grid-area: projects;
}

.palettes {
  height: 80px;
  text-align: center;
}

.palettes>div>button, .palettes>div>input, .palettes>div>select{
  height: 50px;
  margin-left: 10px;
  margin-right: 10px;
  font-weight: bold;
  font-size: 24px;
}

.palettes>div>button{
    background: -webkit-linear-gradient(-180deg, rgb(251, 112, 30), rgb(229, 2, 212), rgb(95, 6, 192));
    color: white;
  }

  .palettes>div>button:hover {
    cursor: pointer;
    box-shadow: 2px 3px 7px #0000008f;
  }


.colors {
  border: 1px solid black;
  border-radius: 250px;
  width: 15%;
  margin: 10px;
  padding-top: 35vh;
  text-align: center;
}

.color-div {
  height: 20px;
  width: 20px;
  border: 1px solid black;
  margin: 5px;
}

.palette-area {
  display: flex;
}

.lock {
  width: 30%;
  position: relative;
  top: 9px;
}

.palette-name {
  border: 0px;
  border-bottom: 1px solid black;
}

.delete {
  width: 10%;
}
