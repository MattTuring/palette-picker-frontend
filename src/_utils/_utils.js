export const getRandomColor = () => {
  let number = '#'+Math.floor(Math.random()*16777215).toString(16);
  if (number.length < 7) number = '#ffffff';
  if (number.length > 7) number = '#000000';
  return number;
}
