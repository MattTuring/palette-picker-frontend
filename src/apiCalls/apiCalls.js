export const getProjects = async () => {
  const url = 'https://palette-pick-backend.herokuapp.com/api/v1/projects';
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error('There was an error getting projects.');
  }
  const projects = await response.json();
  return projects
}

export const getPalettes = async (id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/projects/${id}/palettes`
  const response = await fetch(url);
  if (!response.ok) {
    throw new Error('There was an error getting palettes.');
  }
  const palettes = await response.json();
  return palettes
}

export const postProject = async (project) => {
  const url = 'https://palette-pick-backend.herokuapp.com/api/v1/projects';
  const options = {
    method: 'POST',
    body: JSON.stringify(project),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error posting project.');
  }

  const result = await response.json();
  return result;
}

export const postPalette = async (palette) => {
  const url = 'https://palette-pick-backend.herokuapp.com/api/v1/palettes';
  const options = {
    method: 'POST',
    body: JSON.stringify(palette),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error posting palette.');
  }

  const result = await response.json();
  return result;
}

export const updateProject = async (project, id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/projects/${id}`;
  const options = {
    method: 'PUT',
    body: JSON.stringify(project),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error updating project.');
  }

  const result = await response.json();
  return result;
}

export const updatePalette = async (palette, id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/palettes/${id}`;
  const options = {
    method: 'PUT',
    body: JSON.stringify(palette),
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error updating palette.');
  }

  const result = await response.json();
  return result;
}

export const deleteProject = async (id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/projects/${id}`;
  const options = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error deleting project.');
  }

  const result = await response.json();
  return result;
}

export const deletePalette = async (id) => {
  const url = `https://palette-pick-backend.herokuapp.com/api/v1/palettes/${id}`;
  const options = {
    method: 'DELETE',
    headers: {
      'Content-Type': 'application/json'
    }
  }
  const response = await fetch(url, options);

  if (!response.ok) {
    throw new Error('There was an error deleting palette.');
  }

  const result = await response.json();
  return result;
}
