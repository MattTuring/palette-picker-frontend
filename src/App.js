import React, {useState, useEffect} from 'react';
import './App.css';
import lock from './lock.png'
import unlocked from './open-lock.png'


const randomColor = () => {
  return '#'+Math.floor(Math.random()*16777215).toString(16)
}

const postPalette = (body) => {
  fetch('https://palette-pick-backend.herokuapp.com/api/v1/palettes', {method: 'POST', headers: new Headers({'content-type': 'application/json'}), body: JSON.stringify(body)})
  .then(data => console.log(data))
  .catch(error => console.log(error))
}

const projects = (projects) => {

}

function App() {


   let [locked1, setLocked1] = useState({locked: false})
   let [locked2, setLocked2] = useState({locked: false})
   let [locked3, setLocked3] = useState({locked: false})
   let [locked4, setLocked4] = useState({locked: false})
   let [locked5, setLocked5] = useState({locked: false})


   let [color1, setColor1] = useState({color1: randomColor()})
   let [color2, setColor2] = useState({color2: randomColor()})
   let [color3, setColor3] = useState({color3: randomColor()})
   let [color4, setColor4] = useState({color4: randomColor()})
   let [color5, setColor5] = useState({color5: randomColor()})

   let [projects, setProjects] = useState({})

   let [currentProject, setCurrentProject] = useState({})

   let [palettes, setPalettes] = useState({})

   let [paletteName, setPaletteName] = useState({})


   useEffect(() => {
     fetch('https://palette-pick-backend.herokuapp.com/api/v1/projects')
     .then(results => results.json())
     .then(data => {setProjects(data); setCurrentProject(data[0].id)})

     fetch('https://palette-pick-backend.herokuapp.com/api/v1/palettes')
     .then(results => results.json())
     .then(data => {setPalettes(data)})

   }, [])


  return (
    <main className="App">

    <nav>COLORATOR</nav>

    <section className="color-picker">
      <div style={{backgroundColor: color1.color1}} className="colors color1">{color1.color1} <img className='lock' onClick={() => setLocked1({locked: !locked1.locked})} src={locked1.locked ? lock : unlocked} alt='lock'/></div>
      <div style={{backgroundColor: color2.color2}} className="colors color2">{color2.color2} <img className='lock' onClick={() => setLocked2({locked: !locked2.locked})} src={locked2.locked ? lock : unlocked} alt='lock'/></div>
      <div style={{backgroundColor: color3.color3}} className="colors color3">{color3.color3} <img className='lock' onClick={() => setLocked3({locked: !locked3.locked})} src={locked3.locked ? lock : unlocked} alt='lock'/></div>
      <div style={{backgroundColor: color4.color4}} className="colors color4">{color4.color4} <img className='lock' onClick={() => setLocked4({locked: !locked4.locked})} src={locked4.locked ? lock : unlocked} alt='lock'/></div>
      <div style={{backgroundColor: color5.color5}} className="colors color5">{color5.color5} <img className='lock' onClick={() => setLocked5({locked: !locked5.locked})} src={locked5.locked ? lock : unlocked} alt='lock'/></div>
    </section>

    <section className="projects">
      <h2>Projects</h2>
        {projects.length && projects.map(project => {return <><h4 key={project.id}>{project.name}</h4><button id={project.id}>Show Palettes</button></>})}
    </section>

    <section className="palettes">
      <div>
        <button onClick={() => {
          if (!locked1.locked) {setColor1({color1: randomColor()})}
          if (!locked2.locked) {setColor2({color2: randomColor()})}
          if (!locked3.locked) {setColor3({color3: randomColor()})}
          if (!locked4.locked) {setColor4({color4: randomColor()})}
          if (!locked5.locked) {setColor5({color5: randomColor()})}
        }}>New Colors</button>

        <input className="palette-name" onChange={(event) => {setPaletteName(event.target.value)}} placeholder="Name" type="text"/>

        <select onChange={(event) => {setCurrentProject(event.target.value)}}>
          {projects.length && projects.map(project => {return <option value={project.id} key={project.id + project.id + Math.random}>{project.name}</option>})}
        </select>

        <button onClick={() => postPalette({name: paletteName, color1: color1.color1, color2: color2.color2, color3: color3.color3, color4: color4.color4, color5: color5.color5, project_id: currentProject})}>Add Palette</button>
      </div>
    </section>

    </main>
  );
}

export default App;
