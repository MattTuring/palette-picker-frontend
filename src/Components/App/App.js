import React, { useState, useEffect } from 'react';
import './App.css';
import { getProjects, postPalette } from '../../apiCalls/apiCalls';
import { getRandomColor } from '../../_utils/_utils';
import ProjectsBlock from '../ProjectsBlock/ProjectsBlock';
import PaletteColors from '../PaletteColors/PaletteColors';
import AddPaletteForm from '../AddPaletteForm/AddPaletteForm';

function App() {
  let [locked1, setLocked1] = useState(false);
  let [locked2, setLocked2] = useState(false);
  let [locked3, setLocked3] = useState(false);
  let [locked4, setLocked4] = useState(false);
  let [locked5, setLocked5] = useState(false);

  let [color1, setColor1] = useState('');
  let [color2, setColor2] = useState('');
  let [color3, setColor3] = useState('');
  let [color4, setColor4] = useState('');
  let [color5, setColor5] = useState('');

  let [projects, setProjects] = useState([]);
  let [currentProject, setCurrentProject] = useState({});
  let [paletteName, setPaletteName] = useState({});

  useEffect(() => fetchProjects(), []);
  useEffect(() => assignToRandomColor(), []);

  const assignToRandomColor = () => {
    if (!locked1) setColor1(getRandomColor());
    if (!locked2) setColor2(getRandomColor());
    if (!locked3) setColor3(getRandomColor());
    if (!locked4) setColor4(getRandomColor());
    if (!locked5) setColor5(getRandomColor());
  }

  const fetchProjects = async () => {
    try {
      const recievedProjects = await getProjects();
      setProjects(recievedProjects);
      setCurrentProject(recievedProjects[0].id);
    }
    catch (error) { console.log(error); }
  }

  const addPalette = async () => {
    try {
      const newPalette = {
        name: paletteName,
        color1, color2, color3, color4, color5,
        project_id: currentProject
      };
      await postPalette(newPalette);
    }
    catch (error) { console.log(error); }
  }

  return (
    <main className="App">
      <nav>COLORATOR</nav>
      <PaletteColors
        colors={{color1, color2, color3, color4, color5}}
        lockeds={{locked1, locked2, locked3, locked4, locked5}}
        setLockeds={{setLocked1, setLocked2, setLocked3, setLocked4, setLocked5}} />
      <ProjectsBlock {...{projects, setProjects}} />
      <AddPaletteForm
        {...{assignToRandomColor, paletteName, setPaletteName, setCurrentProject, addPalette, projects}}
        colors={{color1, color2, color3, color4, color5}} />
    </main>
  );
}

export default App;
