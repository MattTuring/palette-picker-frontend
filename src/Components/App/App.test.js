import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { getProjects } from '../../apiCalls/apiCalls';
import { getRandomColor } from '../../_utils/_utils';

jest.mock('../../apiCalls/apiCalls');
jest.mock('../../_utils/_utils');

describe('App', () => {
  let app;

  beforeEach(() => {
    app = shallow(<App />);
  });

  it('should match the snapshot with all data passed in correctly', () => {
    expect(app).toMatchSnapshot();
  });
});
