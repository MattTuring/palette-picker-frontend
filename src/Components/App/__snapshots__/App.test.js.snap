// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<main
  className="App"
>
  <nav>
    COLORATOR
  </nav>
  <PaletteColors
    colors={
      Object {
        "color1": "",
        "color2": "",
        "color3": "",
        "color4": "",
        "color5": "",
      }
    }
    lockeds={
      Object {
        "locked1": false,
        "locked2": false,
        "locked3": false,
        "locked4": false,
        "locked5": false,
      }
    }
    setLockeds={
      Object {
        "setLocked1": [Function],
        "setLocked2": [Function],
        "setLocked3": [Function],
        "setLocked4": [Function],
        "setLocked5": [Function],
      }
    }
  />
  <ProjectsBlock
    projects={Array []}
    setProjects={[Function]}
  />
  <AddPaletteForm
    addPalette={[Function]}
    assignToRandomColor={[Function]}
    colors={
      Object {
        "color1": "",
        "color2": "",
        "color3": "",
        "color4": "",
        "color5": "",
      }
    }
    paletteName={Object {}}
    projects={Array []}
    setCurrentProject={[Function]}
    setPaletteName={[Function]}
  />
</main>
`;
